### ISSUE TITLE
[Rootful] IPv6 DNAT with bound port not forwarding packets

### ISSUE THREAD

**Issue by JCBird1012** [containers/netavark#636]
Hello!

I've been experiencing an issue that feels vaguely similar to containers/podman#7415 (though the reporter in that ticket is using CNI) - using a **rootful** container with an IPv6 bound port seems to not work correctly. I can see the DNAT rule created in `iptables` but it seems that packets never get forwarded.

containers/podman#14709 also looks related - but that's rootless.

### Steps to reproduce (some might be irrelevant, but just to be sure) - all commands run as `root`:

1. Create an custom bridge network with `--ipv6`
```
[root@podman-ipv6-test ~]# podman network create ipv6-test --ipv6
ipv6-test
```

2. Create a rootful container, binding to an IPv6 port on the host
```
[root@podman-ipv6-test ~]# podman run -it -d --network=ipv6-test -p '[::]:8000:8000' python:alpine python -m http.server --bind ::
65c71ac5bcb3ac3a1001c8d9b095eef0bf6860e1a7ebba0bab6667e14429879c
```

3. Double check port is actually bound (well that conmon is holding it), and that DNAT rules exist in `iptables`
```
[root@podman-ipv6-test ~]# sudo netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
...
...
...
tcp6       0      0 :::8000                 :::*                    LISTEN      797/conmon
...
...
...

[root@podman-ipv6-test ~]# ip6tables -t nat -L -n -v
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-DNAT  0    --  *      *       ::/0                 ::/0                 ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 1 packets, 144 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-DNAT  0    --  *      *       ::/0                 ::/0                 ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 1 packets, 144 bytes)
 pkts bytes target     prot opt in     out     source               destination
    1   144 NETAVARK-HOSTPORT-MASQ  0    --  *      *       ::/0                 ::/0
    0     0 NETAVARK-5BDDF4F5C6B6B  0    --  *      *       fdb9:4d85:aa88:8134::/64  ::/0

Chain NETAVARK-5BDDF4F5C6B6B (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 ACCEPT     0    --  *      *       ::/0                 fdb9:4d85:aa88:8134::/64
    0     0 MASQUERADE  0    --  *      *       ::/0                !ff00::/8

Chain NETAVARK-DN-5BDDF4F5C6B6B (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       fdb9:4d85:aa88:8134::/64  ::                   tcp dpt:8000
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       ::1                  ::                   tcp dpt:8000
    0     0 DNAT       6    --  *      *       ::/0                 ::                   tcp dpt:8000 to:[fdb9:4d85:aa88:8134::2]:8000

Chain NETAVARK-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-DN-5BDDF4F5C6B6B  6    --  *      *       ::/0                 ::/0                 tcp dpt:8000 /* dnat name: ipv6-test id: 65c71ac5bcb3ac3a1001c8d9b095eef0bf6860e1a7ebba0bab6667e14429879c */

Chain NETAVARK-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MASQUERADE  0    --  *      *       ::/0                 ::/0                 /* netavark portfw masq mark */ mark match 0x2000/0x2000

Chain NETAVARK-HOSTPORT-SETMARK (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MARK       0    --  *      *       ::/0                 ::/0                 MARK or 0x2000
```

4. Try to access the port using the host's public IPv6 address (from either the host or another machine)
```
Jonathan@petrichor:~$ curl -v6 '[2001:bc8:610:1901::1]:8000'
*   Trying [2001:bc8:610:1901::1]:8000...
* Connected to 2001:bc8:610:1901::1 (2001:bc8:610:1901::1) port 8000 (#0)
> GET / HTTP/1.1
> Host: [2001:bc8:610:1901::1]:8000
> User-Agent: curl/7.86.0
> Accept: */*
>
```

From the host itself - 
```
[root@podman-ipv6-test ~]# curl -v6 'http://[2001:bc8:610:1901::1]:8000'
*   Trying [2001:bc8:610:1901::1]:8000...
* Connected to 2001:bc8:610:1901::1 (2001:bc8:610:1901::1) port 8000 (#0)
> GET / HTTP/1.1
> Host: [2001:bc8:610:1901::1]:8000
> User-Agent: curl/7.86.0
> Accept: */*
>
```

### Some interesting things to take note of...

1. The port is definitely open and listening - curl isn't complaining about connection refused.
2. Using the container's internal IPv6 address on the custom bridge network works as expected
```
[root@podman-ipv6-test ~]# podman container inspect 65c71ac5bcb3ac3a1001c8d9b095eef0bf6860e1a7ebba0bab6667e14429879c
[
...
...
...
               "Networks": {
                    "ipv6-test": {
                         "EndpointID": "",
                         "Gateway": "10.89.0.1",
                         "IPAddress": "10.89.0.2",
                         "IPPrefixLen": 24,
                         "IPv6Gateway": "fdb9:4d85:aa88:8134::1",
                         "GlobalIPv6Address": "fdb9:4d85:aa88:8134::2",
                         "GlobalIPv6PrefixLen": 64,
                         "MacAddress": "be:89:2c:a0:f6:c3",
                         "NetworkID": "ipv6-test",
                         "DriverOpts": null,
                         "IPAMConfig": null,
                         "Links": null,
                         "Aliases": [
                              "65c71ac5bcb3"
                         ]
                    }
               }
...
...
...
]

[root@podman-ipv6-test ~]# curl -v6 'http://[fdb9:4d85:aa88:8134::2]:8000'
*   Trying [fdb9:4d85:aa88:8134::2]:8000...
* Connected to fdb9:4d85:aa88:8134::2 (fdb9:4d85:aa88:8134::2) port 8000 (#0)
> GET / HTTP/1.1
> Host: [fdb9:4d85:aa88:8134::2]:8000
> User-Agent: curl/8.0.1
> Accept: */*
>
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Server: SimpleHTTP/0.6 Python/3.11.2
< Date: Tue, 21 Mar 2023 13:41:36 GMT
< Content-type: text/html; charset=utf-8
< Content-Length: 760
<
...
...
...
* Closing connection 0
```
3. iptables rules appear to be matching correctly - DNAT sees inbound packets and the number changes after curling external IPv6 address
```
[root@podman-ipv6-test ~]# ip6tables -t nat -L -n -v 
...
...
...
Chain NETAVARK-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination
    2   168 NETAVARK-DN-5BDDF4F5C6B6B  6    --  *      *       ::/0                 ::/0                 tcp dpt:8000 /* dnat name: ipv6-test id: 65c71ac5bcb3ac3a1001c8d9b095eef0bf6860e1a7ebba0bab6667e14429879c */
...
...
...

[root@podman-ipv6-test ~]# curl -v6 [2001:bc8:610:1901::1]:8000
*   Trying [2001:bc8:610:1901::1]:8000...
* Connected to 2001:bc8:610:1901::1 (2001:bc8:610:1901::1) port 8000 (#0)
> GET / HTTP/1.1
> Host: [2001:bc8:610:1901::1]:8000
> User-Agent: curl/8.0.1
> Accept: */*
>

[root@podman-ipv6-test ~]# ip6tables -t nat -L -n -v 
...
...
...
Chain NETAVARK-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination
    3   248 NETAVARK-DN-5BDDF4F5C6B6B  6    --  *      *       ::/0                 ::/0                 tcp dpt:8000 /* dnat name: ipv6-test id: 65c71ac5bcb3ac3a1001c8d9b095eef0bf6860e1a7ebba0bab6667e14429879c */
...
...
...
```
4. Running `netstat -tulpn` shows packets stuck in Recv-Q
```
[root@podman-ipv6-test ~]# netstat -tulpn
...
...
...
tcp6       4      0 :::8000                 :::*                    LISTEN      797/conmon
...
...
...
```
### Podman Info
```
[root@podman-ipv6-test ~]# podman system info
host:
  arch: amd64
  buildahVersion: 1.29.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.1.7-1
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: f633919178f6c8ee4fb41b848a056ec33f8d707d'
  cpuUtilization:
    idlePercent: 99.88
    systemPercent: 0.07
    userPercent: 0.05
  cpus: 8
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: podman-ipv6-test
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.2.7-arch1-1
  linkmode: dynamic
  logDriver: journald
  memFree: 33216806912
  memTotal: 33654423552
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 1.8.1-1
    path: /usr/bin/crun
    version: |-
      crun version 1.8.1
      commit: f8a096be060b22ccd3d5f3ebe44108517fbf6c30
      rundir: /run/user/0/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.2.0-1
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 536866816
  swapTotal: 536866816
  uptime: 0h 18m 28.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 1
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 9884618752
  graphRootUsed: 1710727168
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.4.2
  Built: 1677255177
  BuiltTime: Fri Feb 24 16:12:57 2023
  GitCommit: 74afe26887f814d1c39925a1624851ef3590e79c-dirty
  GoVersion: go1.20.1
  Os: linux
  OsArch: linux/amd64
  Version: 4.4.2
  ```

From everything I've read/seen, NAT on IPv6 should work, right? I'd rather not give these containers publicly routable IPv6 addresses - regardless, when port binding, that shouldn't matter.  

This is the third machine I've tried - this test machine is a clean install of Arch Linux - no other iptables rules exist (besides the default), no firewall.

**Comment by Luap99** [comment]
Note that podman/conmon binding the ports shouldn't matter at all. This is only done to prevent other processes from trying to bind them. The iptables rules should always take effect over it.

Does it work when you use `-p 8000:8000` instead? This should forward both ipv4 and ipv6.

**Comment by JCBird1012** [comment]
>  but it doesn’t even look like IPv6 gets bound

Well, clarification on this claim... I should have said conmon doesn't appear bound to the IPv6 port - this shouldn't matter, but it was a red herring that threw me off - the DNAT iptables rules still appear to be in place.

```
[root@podman-ipv6-test ~]# ip6tables -t nat -L -n -v
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

[root@podman-ipv6-test ~]# podman ps -a
CONTAINER ID  IMAGE       COMMAND     CREATED     STATUS      PORTS       NAMES

[root@podman-ipv6-test ~]# podman run -it -d --network=ipv6-test -p 8000:8000 python:alpine python -m http.server --bind ::
6fe179b2f70aee0edabcdfd95d481162ea6fbb27ad2d644c454d75a67797f28c

# Note how conmon isn't bound to :::8080
[root@podman-ipv6-test ~]# sudo netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      346/systemd-resolve
tcp        0      0 127.0.0.54:53           0.0.0.0:*               LISTEN      346/systemd-resolve
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      575/sshd: /usr/bin/
**tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      781/conmon**
tcp        0      0 0.0.0.0:5355            0.0.0.0:*               LISTEN      346/systemd-resolve
tcp6       0      0 :::22                   :::*                    LISTEN      575/sshd: /usr/bin/
tcp6       0      0 :::5355                 :::*                    LISTEN      346/systemd-resolve
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           346/systemd-resolve
udp        0      0 0.0.0.0:5355            0.0.0.0:*                           346/systemd-resolve
udp        0      0 10.89.0.1:53            0.0.0.0:*                           757/aardvark-dns
udp        0      0 127.0.0.54:53           0.0.0.0:*                           346/systemd-resolve
udp        0      0 127.0.0.53:53           0.0.0.0:*                           346/systemd-resolve
udp        0      0 10.75.4.3:68            0.0.0.0:*                           397/systemd-network
udp6       0      0 :::5353                 :::*                                346/systemd-resolve
udp6       0      0 :::5355                 :::*                                346/systemd-resolve
udp6       0      0 fdb9:4d85:aa88:8134::53 :::*                                757/aardvark-dns

[root@podman-ipv6-test ~]# ip6tables -t nat -L -n -v
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-DNAT  0    --  *      *       ::/0                 ::/0                 ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 4 packets, 424 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-DNAT  0    --  *      *       ::/0                 ::/0                 ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 4 packets, 424 bytes)
 pkts bytes target     prot opt in     out     source               destination
    4   424 NETAVARK-HOSTPORT-MASQ  0    --  *      *       ::/0                 ::/0
    0     0 NETAVARK-5BDDF4F5C6B6B  0    --  *      *       fdb9:4d85:aa88:8134::/64  ::/0

Chain NETAVARK-5BDDF4F5C6B6B (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 ACCEPT     0    --  *      *       ::/0                 fdb9:4d85:aa88:8134::/64
    0     0 MASQUERADE  0    --  *      *       ::/0                !ff00::/8

Chain NETAVARK-DN-5BDDF4F5C6B6B (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       fdb9:4d85:aa88:8134::/64  ::/0                 tcp dpt:8000
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       ::1                  ::/0                 tcp dpt:8000
    0     0 DNAT       6    --  *      *       ::/0                 ::/0                 tcp dpt:8000 to:[fdb9:4d85:aa88:8134::2]:8000

Chain NETAVARK-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-DN-5BDDF4F5C6B6B  6    --  *      *       ::/0                 ::/0                 tcp dpt:8000 /* dnat name: ipv6-test id: 6fe179b2f70aee0edabcdfd95d481162ea6fbb27ad2d644c454d75a67797f28c */

Chain NETAVARK-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MASQUERADE  0    --  *      *       ::/0                 ::/0                 /* netavark portfw masq mark */ mark match 0x2000/0x2000

Chain NETAVARK-HOSTPORT-SETMARK (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MARK       0    --  *      *       ::/0                 ::/0                 MARK or 0x2000
```

**Comment by Luap99** [comment]
So am I right that just using `-p 8000:8000` works correctly?

I think I understand what is going on, we assume we get an empty string for all addresses. While we could also get `0.0.0.0` or `::`. Is such cases we cannot treat them as literal ip but instead of to add `/0` to match everything.

compare the destination address of the DNAT rules it is `::` but it should be `::/0`

**Comment by Luap99** [comment]
Fell free to test https://github.com/containers/netavark/pull/638 if able.

**Comment by JCBird1012** [comment]
@Luap99 - #638 appears to resolve my issue - binding works correctly even with `0.0.0.0` and `[::]` set. Thanks for the help.

**Comment by JCBird1012** [comment]
@Luap99 I'm reopening this - I can reproduce something closer to the cause of my original issue. Even your fix in #638 doesn't seem to solve this. 
If you'd prefer I'd turn this into new issue, I'm happy to do that.

Netavark seems to fail to add DNAT rules on the IPv6 side when binding multiple ports. I'm running Netavark built from your branch in the steps below, but this happens on mainline too (so it probably wasn't a regression due to your fix)...

### Steps to Reproduce

```
[root@scw-lucid-cori ~]# /usr/lib/podman/netavark -V
netavark 1.6.0-dev
```

1. Create custom IPv6 bridge network
```
[root@scw-lucid-cori ~]# podman network create ipv6 --ipv6
ipv6
```

2. Run a container with an app that listens on multiple ports... I chose `nginx` here.
```
[root@scw-lucid-cori nginx]# cat nginx.conf
worker_processes  auto;

error_log  /dev/null;
pid        /var/run/nginx.pid;

events {
    worker_connections  4096;
    multi_accept on;
    use epoll;
}


http {
    sendfile on;
    tcp_nopush on;
    tcp_nodelay on;

    gzip off;
    server {
      listen [::]:80 ipv6only=off;
      return 200 'Hello!';
    }

    server {
      listen [::]:443 ipv6only=off;
      return 200 'Hello!';
  }
}

[root@scw-lucid-cori nginx]# podman run -d --network=ipv6 -p 80:80 -p 443:443 -v './nginx.conf:/etc/nginx/nginx.conf' nginx
bef36518be2e69f1da2e3bd1987b8f87ac162f20413098f621d1058a4b79c0a1
```

3. Check the bound ports (I'm using the host's external IPs here)
```
[root@scw-lucid-cori nginx]# curl -v4 163.172.153.244:80
*   Trying 163.172.153.244:80...
* Connected to 163.172.153.244 (163.172.153.244) port 80 (#0)
> GET / HTTP/1.1
> Host: 163.172.153.244
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.23.3
< Date: Thu, 23 Mar 2023 01:25:12 GMT
< Content-Type: text/plain
< Content-Length: 6
< Connection: keep-alive
<
* Connection #0 to host 163.172.153.244 left intact
Hello!

[root@scw-lucid-cori nginx]# curl -v4 163.172.153.244:443
*   Trying 163.172.153.244:443...
* Connected to 163.172.153.244 (163.172.153.244) port 443 (#0)
> GET / HTTP/1.1
> Host: 163.172.153.244:443
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.23.3
< Date: Thu, 23 Mar 2023 01:25:17 GMT
< Content-Type: text/plain
< Content-Length: 6
< Connection: keep-alive
<
* Connection #0 to host 163.172.153.244 left intact
Hello!

[root@scw-lucid-cori nginx]# curl -v6 '[2001:bc8:700:8911::1]:80'
*   Trying [2001:bc8:700:8911::1]:80...
* Connected to 2001:bc8:700:8911::1 (2001:bc8:700:8911::1) port 80 (#0)
> GET / HTTP/1.1
> Host: [2001:bc8:700:8911::1]
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.23.3
< Date: Thu, 23 Mar 2023 01:25:41 GMT
< Content-Type: text/plain
< Content-Length: 6
< Connection: keep-alive
<
* Connection #0 to host 2001:bc8:700:8911::1 left intact
Hello!

[root@scw-lucid-cori nginx]# curl -v6 '[2001:bc8:700:8911::1]:443'
*   Trying [2001:bc8:700:8911::1]:443...
* connect to 2001:bc8:700:8911::1 port 443 failed: Connection refused
* Failed to connect to 2001:bc8:700:8911::1 port 443 after 0 ms: Couldn't connect to server
* Closing connection 0
curl: (7) Failed to connect to 2001:bc8:700:8911::1 port 443 after 0 ms: Couldn't connect to server
```

Here's truncated output of `ip6tables -t nat -L -v -n`:
```
Chain NETAVARK-DN-26815A4C67D60 (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       fdab:ba4:786a:be0e::/64  ::/0                 tcp dpt:80
    0     0 NETAVARK-HOSTPORT-SETMARK  6    --  *      *       ::1                  ::/0                 tcp dpt:80
    1    80 DNAT       6    --  *      *       ::/0                 ::/0                 tcp dpt:80 to:[fdab:ba4:786a:be0e::2]:80
```

I'd expect to see two DNAT rules there. In fact adding it manually fixes the issue.

```
[root@scw-lucid-cori nginx]# sudo ip6tables -t nat -A NETAVARK-DN-26815A4C67D60 -p tcp -m tcp --dport 443 -j DNAT --to-destination [fdab:ba4:786a:be0e::2]:443


[root@scw-lucid-cori nginx]# curl -v6 '[2001:bc8:700:8911::1]:443'
*   Trying [2001:bc8:700:8911::1]:443...
* Connected to 2001:bc8:700:8911::1 (2001:bc8:700:8911::1) port 443 (#0)
> GET / HTTP/1.1
> Host: [2001:bc8:700:8911::1]:443
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.23.3
< Date: Thu, 23 Mar 2023 01:28:23 GMT
< Content-Type: text/plain
< Content-Length: 6
< Connection: keep-alive
<
* Connection #0 to host 2001:bc8:700:8911::1 left intact
Hello!
```

I think this was what I saw initially in my environment, but when I went to reproduce it, I hit #638 which manifested itself the same way.

**Comment by JCBird1012** [comment]
Also, worth mentioning - it might be a race condition somewhere because that setup has _occasionally_ worked - but I'm not sure under what circumstances.
